### PR TITLE
snippets: Add snippets for nrf70 debug

### PR DESCRIPTION
**Pull request by D-Triveni**
This commit adds the overlays for both the driver and supplicant to debug nrf70.

Fixes SHEL-1836.

### CODE REVIEW

File: snippets/nrf70-supplicant-debug/overlay-nrf70-supplicant-debug.conf [hunk 1]
```diff
@@ -18,4 +18,5 @@ CONFIG_SYS_HEAP_RUNTIME_STATS=y
 CONFIG_LOG=y
 CONFIG_PRINTK=y
 CONFIG_LOG_MODE_IMMEDIATE=y
+CONFIG_NRF_WIFI_RPU_RECOVERY=y
```
Comment (krish2718) [inline snippets/nrf70-supplicant-debug/overlay-nrf70-supplicant-debug.conf:21]: how is this related to WPA supplicant debug? This is a driver feature.

File: snippets/nrf70-driver-debug/overlay-nrf70-driver-debug.conf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+CONFIG_NRF700X_UTIL=y
+CONFIG_NET_L2_WIFI_SHELL=y
+CONFIG_NET_STATISTICS=y
+CONFIG_NET_STATISTICS_WIFI=y
+CONFIG_NET_STATISTICS_USER_API=y
+CONFIG_SYS_HEAP_RUNTIME_STATS=y
+CONFIG_LOG=y
+CONFIG_PRINTK=y
+CONFIG_LOG_MODE_IMMEDIATE=y
+CONFIG_NRF_WIFI_RPU_RECOVERY=y
```
Comment (krish2718) [inline snippets/nrf70-driver-debug/overlay-nrf70-driver-debug.conf:21]: `recovery` is opposite of debugging :)

File: snippets/nrf70-driver-debug/overlay-nrf70-driver-debug.conf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+CONFIG_WIFI_NRF700X_LOG_LEVEL_DBG=y
+CONFIG_NRF_WIFI_LOG_LEVEL_DBG=y
```
Comment (krish2718) [inline snippets/nrf70-driver-debug/overlay-nrf70-driver-debug.conf:23]: These suffer from the same problem as WPA supplicant debugs, too verbose and memory consuming. But on the other hand, don't want to end up with too many snippet variants, may be, below?

1. WPA supplicant debug
2. Driver debug
3. Driver verbose debug (enable BUG debug as well)

**Review by krish2718 — CHANGES_REQUESTED** [review]

**Review by VivekUppunda — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: snippets/nrf70-supplicant-debug/overlay-nrf70-supplicant-debug.conf [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2023 Nordic Semiconductor
+# Copyright (c) 2024 Nordic Semiconductor
```
Comment (krish2718) [inline snippets/nrf70-supplicant-debug/overlay-nrf70-supplicant-debug.conf:2]: CONFIG_NRF700X_UTIL=y is not needed for WPA supplicant debug.

File: snippets/nrf70-driver-verbose-debug/overlay-nrf70-driver-verbose-debug.conf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+CONFIG_NET_L2_WIFI_SHELL=y
+CONFIG_NET_STATISTICS=y
+CONFIG_NET_STATISTICS_WIFI=y
+CONFIG_NET_STATISTICS_USER_API=y
+CONFIG_SYS_HEAP_RUNTIME_STATS=y
+CONFIG_LOG=y
+CONFIG_PRINTK=y
+CONFIG_LOG_MODE_IMMEDIATE=y
+CONFIG_WIFI_NRF700X_LOG_LEVEL_DBG=y
+CONFIG_NRF_WIFI_LOG_LEVEL_DBG=y
```
Comment (krish2718) [inline snippets/nrf70-driver-verbose-debug/overlay-nrf70-driver-verbose-debug.conf:22]: we should bus level debug too, `WIFI_NRF700X_BUS_LOG_LEVEL`

**Review by krish2718 — CHANGES_REQUESTED** [review]
> `CONFIG_LOG_MODE_IMMEDIATE=y`

This is useful to avoid log buffer overflows, but can have impact on timing, may be add this to docs.

Also add it migration guide and changelog, as existing snippet is now removed.

File: CODEOWNERS [hunk 6]
```diff
@@ -236,7 +236,9 @@ Kconfig*                                  @tejlmand
 /snippets/nrf91-modem-trace-ext-flash/    @nrfconnect/ncs-cia
 /snippets/nrf91-modem-trace-uart/         @eivindj-nordic
 /snippets/tfm-enable-share-uart/          @nrfconnect/ncs-cia
-/snippets/nrf70-debug/                    @krish2718 @sachinthegreen
+/snippets/nrf70-driver-verbose-debug/     @krish2718 @sachinthegreen
+/snippets/nrf70-driver-debug/             @krish2718 @sachinthegreen
+/snippets/nrf70-supplicant-debug/         @krish2718 @sachinthegreen
```
Comment (krish2718) [inline CODEOWNERS:241]:
```suggestion
/snippets/wpa-supplicant-debug/         @krish2718 @sachinthegreen
```
This is common code.

**Comment by D-Triveni** [comment]
> > `CONFIG_LOG_MODE_IMMEDIATE=y`
> 
> This is useful to avoid log buffer overflows, but can have impact on timing, may be add this to docs.
> 
> Also add it migration guide and changelog, as existing snippet is now removed.

Updated.

File: doc/nrf/protocols/wifi/debugging.rst [hunk 7]
```diff
@@ -28,22 +28,26 @@ With west
 
 .. code-block:: console
 
-    west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-debug"
+    west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-driver-verbose-debug;wpa-supplicant-debug"
```
Comment (krish2718) [inline doc/nrf/protocols/wifi/debugging.rst:31]: We should have sections to explain and demonstrate various debug snippets, combining them in to a single one won't work most of the times due to memory and defeats the purpose of splitting them up.

File: snippets/wpa-supplicant-debug/overlay-wpa-supplicant-debug.conf [hunk 8]
```diff
@@ -9,13 +9,11 @@ CONFIG_SHELL_STACK_SIZE=4600
...
 CONFIG_SHELL_CMDS_RESIZE=n
-CONFIG_NRF700X_UTIL=y
 CONFIG_NET_L2_WIFI_SHELL=y
 CONFIG_NET_STATISTICS=y
 CONFIG_NET_STATISTICS_WIFI=y
 CONFIG_NET_STATISTICS_USER_API=y
 CONFIG_SYS_HEAP_RUNTIME_STATS=y
 CONFIG_LOG=y
 CONFIG_PRINTK=y
-CONFIG_LOG_MODE_IMMEDIATE=y
```
Comment (krish2718) [inline snippets/wpa-supplicant-debug/overlay-wpa-supplicant-debug.conf:20]: My comment https://github.com/nrfconnect/sdk-nrf/pull/16582#pullrequestreview-2201818511 doesn't ask to remove, it should be enabled, else we loose many logs and won't help in debugging, but add a note in the docs.

File: snippets/wpa-supplicant-debug/overlay-wpa-supplicant-debug.conf [hunk 9]
```diff
@@ -9,13 +9,11 @@ CONFIG_SHELL_STACK_SIZE=4600
 CONFIG_NET_SHELL=y
 CONFIG_SHELL_GETOPT=y
 CONFIG_SHELL_CMDS_RESIZE=n
-CONFIG_NRF700X_UTIL=y
 CONFIG_NET_L2_WIFI_SHELL=y
 CONFIG_NET_STATISTICS=y
 CONFIG_NET_STATISTICS_WIFI=y
 CONFIG_NET_STATISTICS_USER_API=y
 CONFIG_SYS_HEAP_RUNTIME_STATS=y
```
Comment (krish2718) [inline snippets/wpa-supplicant-debug/overlay-wpa-supplicant-debug.conf:16]: these are driver stats, unrelated to WPA supplicant.

File: doc/nrf/protocols/wifi/debugging.rst [hunk 10]
```diff
@@ -26,24 +26,57 @@ For example, to build the :ref:`wifi_shell_sample` sample for the nRF7002 DK wit
 With west
 ---------
 
+To build with driver debug logs enabled.
 .. code-block:: console
 
-    west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-debug"
+   west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-driver-debug"
+
+To build with driver verbose, firmware interfacing and BUS interfacing debug logs enabled.
```
Comment (krish2718) [inline doc/nrf/protocols/wifi/debugging.rst:34]:
```suggestion
To build with driver verbose, firmware interface and BUS interface debug logs enabled.
```

Comment (krish2718) [inline doc/nrf/protocols/wifi/debugging.rst:34]:
```suggestion
To build with driver verbose, firmware interface and BUS interface debug logs enabled.
```

File: doc/nrf/protocols/wifi/debugging.rst [hunk 11]
```diff
@@ -26,24 +26,57 @@ For example, to build the :ref:`wifi_shell_sample` sample for the nRF7002 DK wit
...
+
+.. code-block:: console
+
+   west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-driver-verbose-debug"
+
+To build with supplicant debug logs enabled.
```
Comment (krish2718) [inline doc/nrf/protocols/wifi/debugging.rst:40]:
```suggestion
To build with WPA supplicant debug logs enabled.
```

File: doc/nrf/protocols/wifi/debugging.rst [hunk 12]
```diff
@@ -26,24 +26,57 @@ For example, to build the :ref:`wifi_shell_sample` sample for the nRF7002 DK wit
 With west
 ---------
 
+To build with driver debug logs enabled.
```
Comment (krish2718) [inline doc/nrf/protocols/wifi/debugging.rst:29]: may be add a sections for `driver` (normal and verbose) and `wpa_supplicant`?

File: doc/nrf/protocols/wifi/debugging.rst [hunk 13]
```diff
@@ -15,35 +15,73 @@ Software
...
+With West
+   .. code-block:: console
+
+      west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-driver-verbose-debug"
+
+With CMake
+   .. code-block:: console
+
+      cmake -GNinja -Bbuild -DBOARD=nrf7002dk/nrf5340/cpuapp -Dnrf_wifi_shell_SNIPPET="nrf70-driver-verbose-debug" samples/wifi/shell
+      ninja -C build
```
Comment (richabp) [inline doc/nrf/protocols/wifi/debugging.rst:57]: Same as above.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst [hunk 14]
```diff
@@ -72,3 +72,18 @@ Libraries
...
+
+.. toggle::
+
+   The existing snippet ``nrf70-debug`` has been removed and divided into three sub-snippets as below:
+
+     * Use the ``nrf70-driver-debug`` snippet to enable nrf70 driver debug logs.
+
+     * Use the ``nrf70-driver-verbose-logs`` snippet to enable nrf70 driver, Firmware interface and BUS interface debug logs.
+
+     * Use the ``wpa-supplicant-debug`` snippet to enable supplicant logs.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:89]:
```suggestion
     * ``nrf70-driver-debug`` - To enable the nRF70 driver debug logs.
     * ``nrf70-driver-verbose-logs`` - To enable the nRF70 driver, firmware interface, and BUS interface debug logs.
     * ``wpa-supplicant-debug`` - To enable supplicant logs.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 15]
```diff
@@ -660,3 +660,7 @@ Documentation
 * Added the :ref:`peripheral_sensor_node_shield` page.
 * Separated the instructions about building from :ref:`configure_application` and moved it to a standalone :ref:`building` page.
 * Removed the Advanced building procedures page and moved its contents to the :ref:`building` page.
+
+* Updated:
+
+  * The :ref:`ug_nrf70_developing_debugging` with the new snippets added for nrf70 debug.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:666]:
```suggestion
  * The :ref:`ug_nrf70_developing_debugging` page with the new snippets added for nRF70 debug logs.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by krish2718 — APPROVED** [review]

File: doc/nrf/protocols/wifi/debugging.rst [hunk 16]
```diff
@@ -15,35 +15,80 @@ Software
...
+=================
 
-.. code-block:: console
+To build with driver verbose, firmware interface, and BUS interface debug logs enabled, run the following commands:
 
-    west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="nrf70-debug"
+Basic driver debug
+------------------
+
+With West
```
Comment (richabp) [inline doc/nrf/protocols/wifi/debugging.rst:34]:
```suggestion
With west
^^^^^^^^^
```

File: doc/nrf/protocols/wifi/debugging.rst [hunk 17]
```diff
@@ -15,35 +15,80 @@ Software
...
+ .. code-block:: console
 
-    cmake -GNinja -Bbuild -DBOARD=nrf7002dk/nrf5340/cpuapp -Dnrf_wifi_shell_SNIPPET="nrf70-debug" samples/wifi/shell
+    cmake -GNinja -Bbuild -DBOARD=nrf7002dk/nrf5340/cpuapp -Dnrf_wifi_shell_SNIPPET="nrf70-driver-debug" samples/wifi/shell
     ninja -C build
 
+BUS interface level debug
+-------------------------
+
+With West
```
Comment (richabp) [inline doc/nrf/protocols/wifi/debugging.rst:50]:
```suggestion
With west
^^^^^^^^^
```

File: doc/nrf/protocols/wifi/debugging.rst [hunk 18]
```diff
@@ -15,35 +15,80 @@ Software
...
+
+    cmake -GNinja -Bbuild -DBOARD=nrf7002dk/nrf5340/cpuapp -Dnrf_wifi_shell_SNIPPET="nrf70-driver-verbose-debug" samples/wifi/shell
+    ninja -C build
+
+WPA supplicant debug logs
+=========================
+
+To build with WPA supplicant debug logs enabled.
+
+With West
```
Comment (richabp) [inline doc/nrf/protocols/wifi/debugging.rst:68]:
```suggestion
With west
^^^^^^^^^
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 19]
```diff
@@ -755,3 +755,7 @@ Documentation
 * Removed the Advanced building procedures page and moved its contents to the :ref:`building` page.
 * Restructured the :ref:`app_bootloaders` documentation and combined the DFU and bootloader articles.
   Additionally, created a new bootloader :ref:`bootloader_quick_start`.
+
+* Updated:
+
+  * The :ref:`ug_nrf70_developing_debugging` page with the new snippets added for nRF70 debug logs.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:761]:
```suggestion
  * The :ref:`ug_nrf70_developing_debugging` page with the new snippets added for the nRF70 driver debug logs.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**